tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+	g(t).Expect(err).To(gomega.BeNil(), "Failed uninstalling Traefik Helm release")
+}
+
+func DeleteK8ssandraNamespace(t *testing.T) {
+	deleteNamespace(t, namespace)
+}
+
+func DeleteTraefikNamespace(t *testing.T) {
+	deleteNamespace(t, "traefik")
+}
+
+func deleteNamespace(t *testing.T, ns string) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+	k8s.DeleteNamespace(t, kubectlOptions, ns)
+}
+
+func InstallTraefik(t *testing.T) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", "default")
+	_, err := k8s.GetNamespaceE(t, kubectlOptions, "traefik")
+	if err != nil {
+		// Namespace doesn't exist yet, let's create it
+		options := &helm.Options{KubectlOptions: kubectlOptions}
+
+		// Add traefik repo and update repos
+		helm.RunHelmCommandAndGetOutputE(t, options, "repo", "add", "traefik", "https://helm.traefik.io/traefik")
+		helm.RunHelmCommandAndGetOutputE(t, options, "repo", "update")
+
+		// Deploy traefik
+		// helm install traefik traefik/traefik -n traefik --create-namespace -f docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml
+		valuesPath, _ := filepath.Abs("../../docs/content/en/docs/topics/ingress/traefik/kind-deployment/traefik.values.yaml")
+		_, err = helm.RunHelmCommandAndGetOutputE(t, options, "install", "traefik", "traefik/traefik", "-n", "traefik", "--create-namespace", "-f", valuesPath)
+		g(t).Expect(err).To(gomega.BeNil())
+	}
+}
+
+type credentials struct {
+	username string
+	password string
+}
+
+func CheckKeyspaceExists(t *testing.T, keyspace string) {
+	keyspaces := runCassandraQueryAndGetOutput(t, "describe keyspaces")
+
+	// Check that the keyspace exists in the list of keyspaces
+	g(t).Expect(keyspaces).Should(gomega.ContainSubstring(keyspace))
+}
+
+func WaitForReaperPod(t *testing.T) {
+	WaitForPodWithLabelToBeReady(t, "app.kubernetes.io/managed-by=reaper-operator")
+}
+
+func CheckRowCountInTable(t *testing.T, nbRows int, tableName, keyspaceName string) {
+	output := runCassandraQueryAndGetOutput(t, fmt.Sprintf("SELECT id FROM %s.%s", keyspaceName, tableName))
+
+	// Check that we have the right number of rows
+	g(t).Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("(%d rows)", nbRows)))
+}
+
+func CreateCassandraTable(t *testing.T, tableName, keyspaceName string) {
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE KEYSPACE IF NOT EXISTS %s with replication = {'class':'SimpleStrategy', 'replication_factor':1};", keyspaceName))
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s.%s(id timeuuid PRIMARY KEY, val text);", keyspaceName, tableName))
+}
+
+func LoadRowsInTable(t *testing.T, nbRows int, tableName, keyspaceName string) {
+	for i := 0; i < nbRows; i++ {
+		runCassandraQueryAndGetOutput(t, fmt.Sprintf("INSERT INTO %s.%s(id,val) values(now(), '%d');", keyspaceName, tableName, i))
+	}
+}
+
+// Medusa related functions
+func CreateMedusaSecretWithFile(t *testing.T, secretFile string) {
+	home, _ := os.UserHomeDir()
```

Comment:
I hope this isn't creating files to user's home directory who is running the tests? We don't want to pollute the user's filesystem with stuff.

Reply:
Nope, it's reading from it. This is to allow using `~` notation when referencing secrets that are not stored in the GH repo.